core/src/androidMain/kotlin/Peripheral.kt

```diff
@@ -56,7 +56,10 @@ private const val DISCOVER_SERVICES_RETRIES = 5
 public actual fun CoroutineScope.peripheral(
     advertisement: Advertisement,
     builderAction: PeripheralBuilderAction,
-): Peripheral = peripheral(advertisement.bluetoothDevice, builderAction)
+): Peripheral {
+    advertisement as ScanResultAndroidAdvertisement
```

Comment:
If we require a specific implementation of `Advertisement`, that should be reflected in the function params.

Reply:
`ScanResultAndroidAdvertisement` is `internal`, because I don't want to expose `bluetoothDevice` publicly.
It'd be awesome if an `interface` could have `internal` properties, but I'm not aware of a way to make that work.

It would probably be better to have parameter type be `AndroidAdvertisement` regardless though.

Reply:
I don't immediately have a great solution if we're trying to keep the type `internal`. I'll give it some thought, but it shouldn't block the PR for now.